LBHackney-IT/lbh-fss-public-frontend
# Add back button / set postcode functionality

**Description**
Allow the user to navigate forward and backwards through the app 

**Setup**
`npm install`

**Main pages:**
1. Home (Figma 01-Home)

2a. List services [via home form submit] (Figma 02-) example route: `?postcode=bs20by&service_search`

2b. Category explorer [via home category cards under form] (Figma 04-CategoryExplorer) example route: `?category_explorer=11` (1-5 don't work, data issue)

3. Service detail (Figma 03-ServiceProfile) example route: `?service=7`

**Middle layer pages:**
- Set postcode (top right corner of page 2 or 3 above) [?set_postcode=true]
- Select categories [?select_categories=true]
- Select demographics [filters] [?select_demographics=true]

**Test scenarios**
- Set postcode
  - Home postcode field should be valid to a UK postcode and route you to `?postcode={postcode}&service_search` and present a list of services. Your postcode is now stored in localstorage (use incognito if you want to test without)
  - For as long as your postcode is set, main pages #2 and #3 will present this in the top right corner of the overlaying component
  - On submit it should always take you back to the previous page you were on
  - You can click on that button and it will direct you to a new page to set / change your postcode (or could be accessed directly at `?set_postcode=true`) - on submit will list all services
- Back
  - The rule is: If you're on page #3, it always goes back to #2; and page #2 always goes back to page #1
  - Unless you have directly accessed a service page (`?service=7`) and **have not** previously set your postcode - back will take you to home (page #1). If postcode is set then back will take you to #2a

**Test cases**
- Please mix and match
- Start from Home #1 and navigate all the way in and all the way back
- Start from page #2 and go forward or back
- Set postcode / other filter the middle of journey (expect to be returned to the previous page)
- Navigate through start to end, set postcode (back or submit) and expect to land on the same page
- Start from page #3 (set postcode submit / back), go back, select new service, back etc.

**Ticket**
https://app.clickup.com/t/9e5u5p
https://app.clickup.com/t/82pgd5
